_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+
+These are the three problem areas that containers hope to solve; Ease of management, size, and speed to start. There are others, but for simplicity, we will focus on those three as I feel (IMHO) they are the most relevant to business processes. Containers differ from a traditional virtual machine in one significant way. They do not have an actual Operating System. This may come as a surprise to you, even if you might have dabbled with containers before. Containers instead have an abstraction of an operating system that provides hooks into the host operating system using standards and conventions supplied by the container system.
+
+Docker is the de facto container standard for all practical purposes, but other container technologies do exist. The notion of a standardized container comes from the shipping industry. In shipping, there was no standardization, which made transport difficult because you had to figure out how to hold each item on a boat, train, or truck. Containers provided a standard format for moving goods "you can put whatever you want inside the container so long as it conforms to a standard set of dimensions and can be opened and bolted down in a uniform way." This simplified the logistical process of moving real-world items. In this same way, standardizing the way a software container interacts with its host OS to delegate the responsibility for executing tasks simplifies management of the most crucial part, the application.
+
+## Hosting Containers with AWS ECS Fargate
+
+Now that we have a better idea of what makes a container different from a virtual machine, how does this solve our three problems?
+
+First, it makes management easier by letting us write build scripts to create a stable, repeatable representation of the host needed to do the body of work needed by the application. Second, since an OS ultimately hosts the container, that OS can enforce company policy and security policies, so at worst, any container can only do at most that much. For example, if the host OS only allows traffic inbound from a specific subnet and port, the container cannot override that as it is ultimately bound by the host OS's networking rules.
+
+This kind of management reliability continues into data management, memory management, and any other policies that need to be enforced. This ultimately creates a great deal of business agility as operations can reliably open the doors to developers knowing that they have some established rails. As for size, without the need for a whole OS, your container images are often only a few MB, vs. many GB. On the topic of speed, since there isn't a complete boot cycle, the container can effectively start up as fast as the hosted application can start.
```

Comment:
```suggestion
This kind of management reliability continues into data management, memory management, and any other policies that need to be enforced. Ultimately, the result here is business agility. Operations can open the doors to developers knowing that established guide rails are in place. As for size–our second key problem area–since you no longer need a whole OS, container images are often as small as a few MB vs. many GB. On the topic of speed–our third key problem area–since there isn't a complete boot cycle, the container can effectively start up as fast as the hosted application.
```
conciseness, context

Reply:
Thank you this is much better